native/packages/react-native-bpk-component-image/src/BpkImage.js

```diff
@@ -0,0 +1,178 @@
+  type StyleObj,
+} from 'react-native';
+import BpkSpinner from 'react-native-bpk-component-spinner';
+
+import {
+  animationDurationBase,
+  borderRadiusSm,
+  colorGray300,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const styles = StyleSheet.create({
+  outer: {
+    width: '100%',
+    height: '100%',
+  },
+  outerWithBorderRadius: {
+    borderRadius: borderRadiusSm,
+    overflow: 'hidden',
+  },
+  image: {
+    width: '100%',
+    height: '100%',
+    position: 'absolute',
+  },
+  loadingIndicatorView: {
+    width: '100%',
+    height: '100%',
+    backgroundColor: colorGray300,
+    justifyContent: 'center',
+    alignItems: 'center',
+    position: 'absolute',
+  },
+});
+
+export type Props = {
+  onLoad: ?() => mixed,
+  style: ?StyleObj,
+  alt: string,
+  borderRadius: boolean,
+  inView: boolean,
+  loaded: boolean,
+  source: ImageSourcePropType,
+};
+
+type State = {
+  imageOpacity: number,
+  loadingIndicatorOpacity: number,
+  showLoadingIndicator: boolean,
+};
+
+class BpkImage extends Component<Props, State> {
+  static propTypes = {
+    alt: PropTypes.string.isRequired,
+    source: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
+      .isRequired,
+    borderRadius: PropTypes.bool,
+    inView: PropTypes.bool,
+    loaded: PropTypes.bool,
+    onLoad: PropTypes.func,
+    style: ViewPropTypes.style,
+  };
+
+  static defaultProps = {
+    borderRadius: true,
+    inView: true,
+    loaded: true,
+    onLoad: null,
+    style: null,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      showLoadingIndicator: !props.loaded,
+      imageOpacity: new Animated.Value(props.loaded ? 1 : 0),
+      loadingIndicatorOpacity: new Animated.Value(props.loaded ? 0 : 1),
+    };
+  }
+
+  componentWillReceiveProps = newProps => {
+    if (!this.props.loaded && newProps.loaded) {
+      Animated.timing(this.state.imageOpacity, {
+        toValue: 1,
+        duration: animationDurationBase,
+      }).start();
+      // once the image has finished fading in, we can stop
+      // rendering the spinner and background
+      setTimeout(() => {
+        Animated.timing(this.state.loadingIndicatorOpacity, {
+          toValue: 0,
+          duration: animationDurationBase,
+        }).start();
+      }, animationDurationBase);
+      setTimeout(() => {
+        this.setState({ showLoadingIndicator: false });
+      }, 2 * animationDurationBase);
+    }
+  };
+
+  onImageLoad = () => {
+    if (this.props.onLoad) {
+      this.props.onLoad();
+    }
+  };
+
+  render() {
+    const {
+      style: userStyle,
+      source,
+      alt,
+      inView,
+      borderRadius,
+      loaded,
+      ...rest
+    } = this.props;
+
+    const outerStyle = [styles.outer];
+    if (borderRadius) {
+      outerStyle.push(styles.outerWithBorderRadius);
+    }
+    if (userStyle) {
+      outerStyle.push(userStyle);
+    }
+
+    return (
+      <View style={outerStyle} {...rest}>
```

Comment:
As you suggested, I just checked and we need to put `...rest` on the image component, because then consumers can make use of props from the underlying image component like `blurRadius`.